plugins/plugin-languageserver/che-plugin-languageserver-ide/src/main/java/org/eclipse/che/plugin/languageserver/ide/editor/codeassist/CompletionItemBasedCompletionProposal.java

```diff
@@ -87,32 +89,45 @@
 
   @Override
   public void getAdditionalProposalInfo(final AsyncCallback<Widget> callback) {
-    if (completionItem.getItem().getDocumentation() == null && canResolve()) {
-      resolve()
-          .then(
-              item -> {
-                completionItem = item;
-                resolved = true;
-                callback.onSuccess(createAdditionalInfoWidget());
-              })
-          .catchError(
-              e -> {
-                callback.onFailure(e.getCause());
-              });
-    } else {
-      callback.onSuccess(createAdditionalInfoWidget());
-    }
+    MarkedOverlay.getMarkedPromise()
```

Comment:
Maybe you should provide some sort of abstraction instead using `MarkedOverlay` directly.
It's OK to use `MarkedOverlay` in Orion module, but in this module we shouldn't have direct dependency on Orion .

Reply:
You're right, architecturally, but I would propose to merge it like that and to open an issue for it:
1. I'm on PTO in around 2 hours and I'd like to get this merged
2. Investing in architectural integrity of the gwt ide seems low prio.